### PR TITLE
Making IndexSpace accessible from the device

### PR DESCRIPTION
**Pull request by sslattery**
This pull request adds `KOKKOS_INLINE_FUNCTION` definitions to the `IndexSpace` interface functions and calls these functions within device code in the unit tests.

### CODE REVIEW

**Comment by sslattery** [comment]
Note: this still needs to be tested on GPU - I currently do not have access to a GPU machine while traveling

File: unit_test/tstIndexSpace.hpp [hunk 1]
```diff
@@ -402,11 +402,11 @@ void subviewTest()
...
         Kokkos::HostSpace(), v2 );
-    for ( int i = 0; i < size_i; ++i )
-        for ( int j = 0; j < size_j; ++j )
+    for ( int i = 0; i < is2.extent(0); ++i )
+        for ( int j = 0; j < is2.extent(1); ++j )
         {
-            if ( min_i <= i && i < max_i &&
-                 min_j <= j && j < max_j )
+            if ( is2.range(0).first <= i && i < is2.range(0).second &&
+                 is2.range(1).first <= j && j < is2.range(1).second )
```
Comment (dalg24) [inline unit_test/tstIndexSpace.hpp]: This creates temporaries.  Why did you chose that over `is.min(0) <= i && i < is.max(0) && ...`

Comment (sslattery) [reply unit_test/tstIndexSpace.hpp]: In practice one would not do that. But, it allowed me to modify the test to make sure those functions were in fact callable on the device.

**Comment by rhalver** [comment]
> Note: this still needs to be tested on GPU - I currently do not have access to a GPU machine while traveling

Compiled a CUDA-based installation and ran the included test as well as a small test example written by me. All tests passed and my example executed correctly.

File: unit_test/tstIndexSpace.hpp (hunk 1 above)
Comment (dalg24) [reply unit_test/tstIndexSpace.hpp]: Fair enough.  I realize now this is a not production code but a unit test.

**Pull request merged**